include/seadsa/SpecGraphInfo.hh

```diff
@@ -0,0 +1,56 @@
+/**
+   A pass to replace bottom-up PTA of identified functions with a pre-generated
+   spec
+ */
+#pragma once
+
+#include "llvm/ADT/StringRef.h"
+#include "llvm/Analysis/TargetLibraryInfo.h"
+#include "llvm/Pass.h"
+
+#include "seadsa/AllocWrapInfo.hh"
+#include "seadsa/Local.hh"
+
+#include "llvm/IR/BasicBlock.h"
+#include "llvm/IR/Function.h"
+#include "llvm/IR/Module.h"
+#include "llvm/IR/Value.h"
+#include <unordered_map>
+
+namespace llvm {
+class Function;
+class Module;
+class TargetLibraryInfoWrapperPass;
+class Function;
+} // namespace llvm
+
+namespace seadsa {
+class LocalAnalysis;
+class AllocWrapInfo;
+
+class SpecGraphInfo : public llvm::ImmutablePass {
+
+protected:
+  mutable AllocWrapInfo *m_awi;
+  mutable llvm::TargetLibraryInfoWrapperPass *m_tliWrapper;
+  mutable std::unordered_map<std::string, llvm::Function *> m_funcs;
+
+  // m_module must be deallocated before m_ctx, keep in this order
+  mutable llvm::LLVMContext m_ctx;
+  mutable std::unique_ptr<llvm::Module> m_module;
```

Comment:
should not be hard to extend it to multiple modules. We might have a spec module per library rather than one module that links all specs together.

Reply:
How would that manifest itself? would the user have to provide a directory where all specs are stored, a list of paths, something else?

Reply:
I think the easiest is the option repeated many times as in `--dsa-spec=/ag/spec1.bc --dsa-spec=/ag/spec2.bc`.

For inspiration, think `-l` or `-L` flag for linking libraries in the compiler